src/symbolize/gimli/libs_windows.rs

```diff
@@ -43,13 +42,67 @@ unsafe fn add_loaded_images(ret: &mut Vec<Library>) {
     }
 }
 
+// Safety: long_path should be null-terminated
+#[cfg(target_os = "cygwin")]
+unsafe fn get_posix_path(long_path: &[u16]) -> Option<OsString> {
+    use super::mystd::os::unix::ffi::OsStringExt;
+
+    unsafe extern "C" {
+        fn cygwin_conv_path(
+            what: libc::c_uint,
+            from: *const libc::c_void,
+            to: *mut libc::c_void,
+            size: libc::size_t,
+        ) -> libc::ssize_t;
+    }
+    const CCP_WIN_W_TO_POSIX: libc::c_uint = 3;
+
+    let name_len = unsafe {
+        cygwin_conv_path(
+            CCP_WIN_W_TO_POSIX,
+            long_path.as_ptr().cast(),
+            core::ptr::null_mut(),
+            0,
+        )
+    };
```

Comment:
Please add a comment documenting how `cygwin_conv_path` returns the length you must allocate if you call it with null pointers. And possibly a link to https://cygwin.com/cygwin-api/func-cygwin-conv-path.html as well?

Reply:
Comments are added. The documentation didn't say that `to` is allowed to be null if `size` is 0. The source code of cygwin, however, allows it.

Do you think I need to change `null_mut()` to `1 as *mut c_void`?